### PR TITLE
[Admin] Taxons index

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | N/A
| License         | MIT

image

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/Admin/ManagingTaxonsContext.php [hunk 1]
```diff
@@ -358,23 +358,33 @@ public function taxonNamedShouldNotBeAdded(string $name): void
...
     public function iShouldSeeTheTaxonNamedInTheList(string $name): void
     {
-        Assert::same($this->createPage->countTaxonsByName($name), 1);
+        Assert::true($this->createPage->isTaxonOnTheList($name));
+    }
+
+    /**
+     * @Then the order of taxons should be :firstTaxon, :secondTaxon, :thirdTaxon and :fourthTaxon
+     */
+    public function theOrderOfTaxonsShouldBe(string ...$taxonsName): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingTaxonsContext.php]:
```suggestion
    public function theOrderOfTaxonsShouldBe(string ...$taxonsNames): void
```

File: src/Sylius/Behat/Page/Admin/Taxon/CreatePage.php
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/Taxon/CreatePage.php:1]: Extract methods for managing taxons tree to a separate element

**Review by mpysiak — DISMISSED** [review]

File: src/Sylius/Bundle/AdminBundle/templates/taxon/sections/tree.html.twig
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/taxon/sections/tree.html.twig:1]: Is it possible to divide these buttons into Twig Hooks? 🤔

File: src/Sylius/Behat/Element/Admin/Taxon/TreeElement.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    protected function getDefinedElements(): array
+    {
+        return array_merge(parent::getDefinedElements(), [
+            'tree_taxons' => '[data-test-tree-taxons]',
+            'tree_taxon_actions' => '[data-test-tree-taxons] [data-test-tree-taxon="%name%"] [data-test-actions]',
+            'tree_taxon_move_down' => '[data-test-tree-taxons] [data-test-tree-taxon="%name%"] [data-test-move-down]',
+            'tree_taxon_move_up' => '[data-test-tree-taxons] [data-test-tree-taxon="%name%"] [data-test-move-up]',
+            'first_tree_taxon' => '[data-test-tree-taxons] [data-test-tree-taxon]:first-child',
+            'last_tree_taxon' => '[data-test-tree-taxons] [data-test-tree-taxon]:last-child',
+            'tree_taxon_component' => '[data-live-name-value="sylius_admin:taxon:tree"]',
```
Comment (GSadee) [inline src/Sylius/Behat/Element/Admin/Taxon/TreeElement.php:83]: Not alphabetical order

File: src/Sylius/Behat/Context/Ui/Admin/RemovingTaxonContext.php [hunk 3]
```diff
@@ -49,7 +49,7 @@ public function theTaxonShouldStillExist(TaxonInterface $taxon): void
     {
         $this->createPage->open();
 
-        Assert::true($this->createPage->hasTaxonWithName($taxon->getName()));
+        Assert::true($this->createPage->isTaxonOnTheList($taxon->getName()));
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/RemovingTaxonContext.php:52]:
```suggestion
        Assert::true($this->treeElement->isTaxonOnTheList($taxon->getName()));
```
🤔

**Review by GSadee — APPROVED** [review]

**Pull request merged**